examples/src/main/java/org/apache/spark/examples/streaming/JavaRecoverableNetworkWordCount.java

```diff
@@ -204,13 +202,17 @@ public static void main(String[] args) {
     final int port = Integer.parseInt(args[1]);
     final String checkpointDirectory = args[2];
     final String outputPath = args[3];
-    JavaStreamingContextFactory factory = new JavaStreamingContextFactory() {
+
+    // Function to create JavaStreamingContext without any output operations
+    // (used to detect the new context)
+    Function0<JavaStreamingContext> createContextFunc = new Function0<JavaStreamingContext>() {
```

Comment:
As this is a Java example, I would prefer to leave like this unless you have strong feelings about the function declaration